### PR TITLE
:mortar_board: Lab5 --- linked structures

### PR DESCRIPTION
**Pull request by jameshughes89**
### What
Lab for building linked structures.

### Why
Because they will be using them a bunch.

### CODE REVIEW

File: src/site/lab5.rst [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+
+1. Write a static method in the same class as your ``main`` called ``addToFront`` that will
+    a. Take a reference to the head of a linked structure and an integer to add
+    b. Insert a new node containing reference to a new integer passed to the method as a parameter
+    c. Return a reference to the new head of the linked structure
+
+    .. code-block:: java
+        :linenos:
+
+        public static Node<T> addToFront(Node<T> head, Integer toAdd) {
```
Comment (twentylemon) [inline src/site/lab5.rst:87]: `Integer` -> `T`. occurs in instructions and other methods as well.

File: src/site/lab5.rst [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+
+1. Write a static method in the same class as your ``main`` called ``makeLinkedStructure`` that will
+    a. Create a linked structure containing the numbers 0 -- 9
+        * One number referenced in each node
+    b. Return a reference to the head of the linked structure
+
+    .. code-block:: java
+        :linenos:
+
+        public static Node<T> makeLinkedStructure() {
```
Comment (twentylemon) [inline src/site/lab5.rst:51]: Method needs to be a template method, `public <T> Node<T>`. Occurs for all signatures.

**Review by twentylemon — CHANGES_REQUESTED** [review]

File: src/site/lab5.rst (hunk 1 above)
Comment (jameshughes89) [reply src/site/lab5.rst:87]: Good catch my man.

File: src/site/lab5.rst [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+1. Write a static method in the same class as your ``main`` called ``addToMiddle`` that will
+    a. Take a reference to the head of a linked structure, an integer to add, and a number we want to insert our new number after
+        * For example, if we call ``addToMiddle(head, 99, 5)``, we will add a node containing a reference to ``99`` after the node containing a reference to the number ``5`` (if it exists)
+    b. Insert a new node containing reference to a new integer passed to the method as a parameter after the specified value
+    c. Return a reference to the head of the linked structure
+
+    .. code-block:: java
+        :linenos:
+
+        public static <T> Node<T> addToMiddle(Node<T> head, Integer toAdd, T addAfter) {
```
Comment (twentylemon) [inline src/site/lab5.rst:146]: int -> `T` here still

**Review by twentylemon — APPROVED** [review]

**Pull request merged**